src/shared/Microsoft.Git.CredentialManager/Interop/Posix/Native/Stdlib.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Git.CredentialManager.Interop.Posix.Native
+{
+    public static class Stdlib
+    {
+        [DllImport("libc", CallingConvention = CallingConvention.Cdecl, CharSet = CharSet.Ansi)]
+        public static extern unsafe IntPtr realpath(byte* path, byte* resolved_path);

```

Comment:
How are we doing error handling here? I only made it through the test code, but it looks like we are not checking the actual error? (Either with `SetLastError` or `Marshal.GetLastWin32Error`)?

Reply:
We are not. Not sure `Marshal.GetLastWin32Error` will work here on this POSIX call.

This is really only used in tests, and then only to resolve the symlinked temporary directory on macOS because.. Apple.. and there is no .NET API exposing 'realpath' because.. Microsoft. 😢

```
lrwxr-xr-x@ 1 root  wheel  11 31 Jan 23:02 /var -> private/var
```

I'm tempted to just remove this, the tests it supports (testing of `LibGit2.GetRepositoryPath` which just wraps `git_repository_discover`) because of this mess with symlinks 😢